src/main.cpp

```diff
@@ -1128,3 +1135,41 @@ static bool recompress_map_database(const GameParams &game_params, const Setting
 	actionstream << "Done, " << count << " blocks were recompressed." << std::endl;
 	return true;
 }
+
+static void get_env_opts(Settings &settings)
+{
+	// allows turning on/off colored output via env (see issue #7553)
+	// CLICOLOR is a de-facto standard option for colors in ANSI-capable environments.
+	// CLICOLOR != 0: ANSI colors are supported
+	// CLICOLOR == 0: ANSI colors are NOT supported
+	const char *clicolor_raw = std::getenv(ENV_CLICOLOR);
+	if (clicolor_raw) {
+		const std::string color = clicolor_raw;
+		if (color == "0") {
+			settings.set("color", "never");
+		} else {
+			// auto since it still could be not a tty
+			settings.set("color", "auto");
+		}
+	}
+	// NO_COLOR only specifies that NO color is allowed - whether or not
+	// we actually give the user color is up to the "auto" mode.
+	// Implemented according to no-color.org (08-2022)
+	const char *no_color_raw = std::getenv(ENV_NO_COLOR);
+	if (no_color_raw) {
+		const std::string color = no_color_raw;
+		if (!color.empty()) {
+			settings.set("color", "never");
+		}
+	}
+	// CLICOLOR_FORCE is another option of the CLICOLOR semi-standard, which
+	// should turn on colors "no matter what".
+	const char *clicolor_force_raw = std::getenv(ENV_CLICOLOR_FORCE);
+	if (clicolor_force_raw) {
+		const std::string color = clicolor_force_raw;
+		if (color != "0") {
+			// should ALWAYS have colors, so we ignore tty (no "auto")
+			settings.set("color", "always");
+		}
+	}
```

Comment:
not critical but this can be simplified somewhat

```suggestion
	// CLICOLOR is a de-facto standard option for colors <https://bixense.com/clicolors/>
	// CLICOLOR != 0: ANSI colors are supported (auto-detection, this is the default)
	// CLICOLOR == 0: ANSI colors are NOT supported
	const char *clicolor = std::getenv(ENV_CLICOLOR);
	if (clicolor && std::string(clicolor) == "0") {
		settings.set("color", "never");
	}
	// NO_COLOR only specifies that no color is allowed.
	// Implemented according to <http://no-color.org/>
	const char *no_color = std::getenv(ENV_NO_COLOR);
	if (no_color && no_color[0]) {
		settings.set("color", "never");
	}
	// CLICOLOR_FORCE is another option, which should turn on colors "no matter what".
	const char *clicolor_force = std::getenv(ENV_CLICOLOR_FORCE);
	if (clicolor_force && std::string(clicolor_force) != "0") {
		// should ALWAYS have colors, so we ignore tty (no "auto")
		settings.set("color", "always");
	}
```